### PR TITLE
[android] #3052 - upgrade gradle to 2.10

### LINKED ISSUE #3052: Upgrade to Gradle 2.9
Gradle 2.9 was just released.

Main features over at https://docs.gradle.org/current/release-notes

Of notable interest is:
> Explicit declaration of library API with the experimental Java software model
> Developing with the experimental Java software model is now more powerful, with the ability to explicitly declare which packages and dependencies make up the API of a library.

This should help us solve the internal method problem we have in the annotation library.

/cc @bleege @tobrun @zugaldia

### PR DESCRIPTION
**Pull request by tobrun**
closes #3052

### CODE REVIEW

File: platform/android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,7 +1,6 @@
-#Thu Nov 05 15:43:10 CST 2015
+#Fri Jan 15 17:23:43 CET 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.8-bin.zip
-distributionSha256Sum=a88db9c2f104defdaa8011c58cf6cda6c114298ae3695ecfb8beb30da3a903cb
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-bin.zip
```
Comment (bleege) [inline platform/android/gradle/wrapper/gradle-wrapper.properties]: We need to bring the distributinoSha256Sun property over here too.  Sadly this is always a hand copy thing.

**Comment by bleege** [comment]
This is what a happy SHA does.  Should be good to merge this when Bitrise :ok_hand: things.

screen shot 2016-01-15 at 12 47 30 pm

**Comment by bleege** [comment]
Merged on to `release-android-3.1.0 branch`.

**Comment by tobrun** [comment]
:thumbsup:

**Pull request merged**